tests/app/forms/test_questionnaire_form.py

```diff
@@ -16,1677 +16,1678 @@
-                "csrf_token": None,
-                "date-range-from-answer": "2016-12-25",
-                "date-range-to-answer": "2016-12-25",
+        form_data = MultiDict(
+            {
+                "date-range-from-answer-day": "25",
+                "date-range-from-answer-month": "12",
+                "date-range-from-answer-year": "2016",
+                "date-range-to-answer-day": "25",
+                "date-range-to-answer-month": "12",
+                "date-range-to-answer-year": "2016",
             }
-            form = generate_form(
-                schema,
-                question_schema,
-                AnswerStore(),
-                ListStore(),
-                metadata=None,
-                response_metadata={},
-                form_data=form_data,
-            )
+        )
 
-            form.validate()
-            self.assertEqual(form.data, expected_form_data)
-            self.assertEqual(
-                form.question_errors["date-range-question"],
-                schema.error_messages["INVALID_DATE_RANGE"],
-            )
+        expected_form_data = {
+            "csrf_token": None,
+            "date-range-from-answer": "2016-12-25",
+            "date-range-to-answer": "2016-12-25",
+        }
+        form = generate_form(
+            schema,
+            question_schema,
+            answer_store,
+            list_store,
+            metadata=None,
+            response_metadata={},
+            form_data=form_data,
+        )
 
-    def test_date_range_to_precedes_from_raises_question_error(self):
-        with self.app_request_context():
-            schema = load_schema_from_name("test_date_range")
+        form.validate()
+        assert form.data, expected_form_data
+        assert (
+            form.question_errors["date-range-question"]
+            == schema.error_messages["INVALID_DATE_RANGE"]
+        )
 
-            question_schema = schema.get_block("date-block").get("question")
 
-            form_data = MultiDict(
-                {
-                    "date-range-from-answer-day": "25",
-                    "date-range-from-answer-month": "12",
-                    "date-range-from-answer-year": "2016",
-                    "date-range-to-answer-day": "24",
-                    "date-range-to-answer-month": "12",
-                    "date-range-to-answer-year": "2016",
-                }
-            )
+def test_date_range_to_precedes_from_raises_question_error(
+    app, answer_store, list_store
+):
+    with app.test_request_context():
+        schema = load_schema_from_name("test_date_range")
 
-            expected_form_data = {
-                "csrf_token": None,
-                "date-range-from-answer": "2016-12-25",
-                "date-range-to-answer": "2016-12-24",
+        question_schema = schema.get_block("date-block").get("question")
+
+        form_data = MultiDict(
+            {
+                "date-range-from-answer-day": "25",
```

Comment:
Consider making them global/fixture/file as they seem to be repeated with same data in couple of tests. I'm ok if you keep it as is.